src/Riak.php

```diff
@@ -213,10 +226,15 @@ public function getLastRequest()
      * @return $this
      * @throws Exception
      */
-    protected function pickNewNode()
+    public function pickNewNode()
     {
-        // mark current active node as inactive
+        // mark current active node as inactive and increment attempts
         $this->getActiveNode()->setInactive(true);
+        $this->attempts++;
+        $this->inactiveNodes[$this->getActiveNodeIndex()] = $this->getActiveNode();
```

Comment:
In which case wouldn't then be better to let that job to a more suited tool? A load balancer maybe or something similar?